tests/FSharp.Compiler.ComponentTests/EmittedIL/Optimization/samples/for_loop_custom_step.fsx

```diff
@@ -0,0 +1,10 @@
+// <testmetadata>
+// { "optimization": { "reported_in": "#9548", "reported_by": "@teo-tsirpanis", "last_know_version_not_optimizing": "8", "first_known_version_optimizing": null } }
+// </testmetadata>
+
+open System
+
+let g() =
+  for x in 0 .. 2 .. 15 do
+    Console.WriteLine x
+g()
```

Comment:
This one and `for_loop_non_int` should now both be optimized after #16650 (with langversion `preview`).

Reply:
Thanks for the heads up! I need to figure out the remaining baselines that fail on CI to hopefully wrap up this PR.